ddpui/services/metric_service.py

```
@@ -0,0 +1,309 @@
+                qb.add_column(literal_column(metric.column_expression).label("metric_value"))
+            else:
+                qb.add_aggregate_column(metric.column, metric.aggregation, alias="metric_value")
+
+            qb.limit_rows(1)
+            sql_stmt = qb.build()
+            compiled = sql_stmt.compile(
+                bind=warehouse.engine, compile_kwargs={"literal_binds": True}
+            )
+            warehouse.execute(compiled)
+            logger.info(f"Metric validation query succeeded for '{metric.name}'")
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)